### PR TITLE
DDK: implement BTI, PMT, VMOContiguous

### PR DESCRIPTION
**Pull request by rzswh**
Implement following feature:
- vmo_create_physical
- vmo_create_contiguous
- bti
- pmt
- create slice by `vmo_create_child` (**Replaced by my code**)
- and so on

### CODE REVIEW

File: zircon-object/src/vm/vmo/paged.rs [hunk 1]
```diff
@@ -512,6 +706,11 @@ impl VMObjectPaged {
                     if let Some(frame) = inner.frames.get(&i) {
                         if frame.tag.is_split() {
                             let mut new_frame = inner.frames.remove(&i).unwrap();
+                            if inner.contiguous && !other_child.contiguous {
```
Comment (PanQL) [inline zircon-object/src/vm/vmo/paged.rs:709]: this `if` statement can be collapsed

File: zircon-object/src/vm/vmo/paged.rs [hunk 2]
```diff
@@ -654,7 +872,18 @@ impl VMObjectPagedInner {
         let start = child.parent_offset / PAGE_SIZE;
         let end = child.parent_limit / PAGE_SIZE;
         // merge nodes to the child
-        for (key, value) in self.frames.split_off(&start) {
+        let mut child_frames = self.frames.split_off(&start);
+        for (key, value) in child_frames.iter_mut() {
+            if *key >= end {
+                break;
+            }
+            if self.contiguous && !child.contiguous {
```
Comment (PanQL) [inline zircon-object/src/vm/vmo/paged.rs:880]: this `if` statement can be collapsed

File: zircon-object/src/vm/vmo/paged.rs [hunk 3]
```diff
@@ -808,6 +1005,11 @@ impl VMObjectPagedInner {
                 if start <= idx && idx < end {
                     if locked_parent.frames.contains_key(&idx) {
                         let mut to_insert = locked_parent.frames.remove(&idx).unwrap();
+                        if locked_parent.contiguous && !locked_other.contiguous {
```
Comment (PanQL) [inline zircon-object/src/vm/vmo/paged.rs:1008]: this `if` statement can be collapsed

File: zircon-syscall/src/channel.rs [hunk 4]
```diff
@@ -32,8 +32,8 @@ impl Syscall<'_> {
         const MAY_DISCARD: u32 = 1;
         let never_discard = options & MAY_DISCARD == 0;
 
-        // let test_args = "test\0-f\0VmoZeroTest*\0";
-        let test_args = "";
+        let test_args = "test\0-f\0VmoClone2TestCase.ContiguousVmoPartialClone\0";
```
Comment (PanQL) [inline zircon-syscall/src/channel.rs:35]: running shell will be failed because of this non-empty test_args

**Review by wangrunji0408 — APPROVED** [review]

**Pull request merged**